src/main/java/com/sac/SalveAmuletCheckerPlugin.java

```diff
@@ -26,249 +25,224 @@
-	}
-
-	public boolean isSalveAmulet(int itemId){
-		boolean isSalveAmulet = false;
-		if(itemId == ItemID.SALVE_AMULET || itemId == ItemID.SALVE_AMULET_E || itemId == ItemID.SALVE_AMULETEI){
-			isSalveAmulet = true;
-		}
-		return isSalveAmulet;
-	}
-
-	private SalveAmuletCheckerInfoBox addInfoBox(BufferedImage image, String text, String toolTip){
-		val infoBox = new SalveAmuletCheckerInfoBox(image,this);
-		infoBox.setInfoBoxText(text);
-		infoBox.setTooltip(toolTip);
-		return infoBox;
-	}
-
-	private void removeInfoBox(InfoBox infoBox){
-		infoBoxManager.removeInfoBox(infoBox);
-		infoBox = null;
-	}
-
-	private SalveAmuletCheckerInfoBox createInfoBox(BufferedImage image, String text, String toolTip, Color textBoxColor){
-		val salveAmuletInfoBox = new SalveAmuletCheckerInfoBox(image,this);
-		salveAmuletInfoBox.setInfoBoxText(text);
-		salveAmuletInfoBox.setTooltip(toolTip);
-		salveAmuletInfoBox.setInfoBoxTextColor(textBoxColor);
-		return salveAmuletInfoBox;
-	}
-
-	private void addCoxSalveAmuletCheckerInfoBox(String text, String toolTip, Color color) {
-		removeInfoBox(locationSalveAmuletInfoBox);
-		locationSalveAmuletInfoBox = addInfoBox(itemManager.getImage(ItemID.SALVE_AMULETEI), text, toolTip);
-		infoBoxManager.addInfoBox(locationSalveAmuletInfoBox);
-	}
-
-	private void addPlayerSalveAmuletCheckerInfoBox(String text, String toolTip, Color color){
-		removeInfoBox(coxSalveAmuletCheckerInfoBox);
-		coxSalveAmuletCheckerInfoBox = addInfoBox(itemManager.getImage(ItemID.DRAGON_2H_SWORD), text, toolTip);
-		infoBoxManager.addInfoBox(coxSalveAmuletCheckerInfoBox);
-	}
-
-	private void addRaidStateInfoBox(String text, String toolTip, Color color){
-		removeInfoBox(raidStateInfoBox);
-		raidStateInfoBox = addInfoBox(itemManager.getImage(ItemID.CLEAR_LIQUID), text, toolTip);
-		infoBoxManager.addInfoBox(raidStateInfoBox);
-	}
+public class SalveAmuletCheckerPlugin extends Plugin {
+    @Inject
+    private Client client;
+
+    @Inject
+    private SalveAmuletCheckerConfig config;
+
+    @Inject
+    private ClientToolbar clientToolbar;
+
+    @Inject
+    public TobManager tobManager;
+
+    @Inject
+    public CoxManager coxManager;
+
+    @Inject
+    private InfoBoxManager infoBoxManager;
+
+    @Inject
+    private ItemManager itemManager;
+
+    @Inject
+    private OverlayManager overlayManager;
+
+    @Inject
+    private MysticRoomOverlay mysticRoomOverlay;
+    @Inject
+    private CoxLocationOverlay coxLocationOverlay;
+    @Inject
+    private BloatRoomOverlay bloatRoomOverlay;
+    @Inject
+    private TobLocationOverlay tobLocationOverlay;
+
+
+    private TobState currentTobState;
+    private SalveAmuletCheckerPanel panel;
+    private NavigationButton navButton;
+
+    private static final Set<Integer> SALVE_AMULET_IDS = Set.of(
+            ItemID.SALVE_AMULET,
+            ItemID.SALVE_AMULET_E,
+            ItemID.SALVE_AMULETEI,
+            ItemID.SALVE_AMULETI_26763,
```

Comment:
Nice find, it looks like these variants are from soul wars and emir's arena, which could explain why it felt so random.